powerstore/resource_volume.go

```diff
@@ -456,3 +563,34 @@ func (r resourceVolume) Delete(ctx context.Context, req tfsdk.DeleteResourceRequ
 	resp.State.RemoveResource(ctx)
 	log.Printf("Done with Delete")
 }
+
+// ImportState import state for existing infrastructure
+func (r resourceVolume) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+
+	log.Printf("Started with Import")
+
+	// fetching Volume information
+	response, err := r.p.client.PStoreClient.GetVolume(context.Background(), req.ID)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error importing Volume",
+			fmt.Sprintf("Could not import volume: %s with error: %s", req.ID, err.Error()),
+		)
+		return
+	}
+
+	state := models.Volume{}
+
+	// as state is like a plan here, a current state prior to this import operation
+	updateVolState(&state, response, nil, gopowerstore.VolumeGroups{}, &state, "import")
```

Comment:
same as above
try to use https://github.com/dell/terraform-provider-powerstore/blob/dev/powerstore/defaults.go#L14

also, try to make updateVolState as method instead of function